src/main/java/org/broadinstitute/hellbender/utils/fermi/FermiLiteAssembly.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.hellbender.utils.fermi;
+
+import java.io.*;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+/** an assembly is just a collection of contigs */
+public final class FermiLiteAssembly {
+    private final List<Contig> contigs;
+
+    FermiLiteAssembly(final List<Contig> contigs ) { this.contigs = contigs; }
+
+    public int getNContigs() { return contigs.size(); }
+    public Contig getContig( final int idx ) { return contigs.get(idx); }
+    public List<Contig> getContigs() { return Collections.unmodifiableList(contigs); }
+
+    /** a sequence of bases, coverage data, and connections to other contigs */
+    public final static class Contig {
+        private final byte[] sequence;
+        private final byte[] perBaseCoverage;
+        private final int nSupportingReads;
+        private List<Connection> connections;
+
+        Contig( final byte[] sequence, final byte[] perBaseCoverage, final int nSupportingReads ) {
+            this.sequence = sequence;
+            this.perBaseCoverage = perBaseCoverage;
+            this.nSupportingReads = nSupportingReads;
+        }
+
+        public byte[] getSequence() { return sequence; }
+        public byte[] getPerBaseCoverage() { return perBaseCoverage; }
+        public int getNSupportingReads() { return nSupportingReads; }
+        public List<Connection> getConnections() { return Collections.unmodifiableList(connections); }
+
+        void setConnections( final List<Connection> connections ) {
+            this.connections = connections;
+        }
+    }
+
+    /** a connection between contigs */
+    public final static class Connection {
+        private final Contig target;      // contig that overlaps the one that possesses this connection
+        private final int overlapLen;     // bases in common -- negative overlap lengths are legal, and represent gaps
+        private final boolean isRC;       // if target is a predecessor
+        private final boolean isTargetRC; // if connection is to RC of target contig
+
+        Connection( final Contig target, final int overlapLen, final boolean isRC, final boolean isTargetRC ) {
+            this.target = target;
+            this.overlapLen = overlapLen;
+            this.isRC = isRC;
+            this.isTargetRC = isTargetRC;
+        }
+
+        public Contig getTarget() { return target; }
+        public int getOverlapLen() { return overlapLen; }
+        public boolean isRC() { return isRC; }
+        public boolean isTargetRC() { return isTargetRC; }
+    }
+
+    public void writeGFA( final OutputStream os ) throws IOException {
```

Comment:
It could be huge.  If you know it's small and want to capture it as a string, use a ByteArrayOutputStream.  That won't be the usual use case.
I'll add a unit test.